### PR TITLE
deps: bump go-getter to 1.5.4

### PR DESCRIPTION
**Pull request by tgross**
Fixes https://github.com/hashicorp/nomad/issues/10438

### CODE REVIEW

File: vendor/github.com/hashicorp/go-cleanhttp/cleanhttp.go [hunk 1]
```diff
@@ -32,6 +32,7 @@ func DefaultPooledTransport() *http.Transport {
 		IdleConnTimeout:       90 * time.Second,
 		TLSHandshakeTimeout:   10 * time.Second,
 		ExpectContinueTimeout: 1 * time.Second,
+		ForceAttemptHTTP2:     true,
```
Comment (notnoop) [inline vendor/github.com/hashicorp/go-cleanhttp/cleanhttp.go:35]: Is switch to http2 worth calling out in changelog?

**Review by notnoop — APPROVED** [review]
LGTM.

The line between improvement vs bug seems very fuzzy to me. Is it worth backporting to 1.0.X? Though, dependency changes curry a bit of risk so it makes sense not to backport if we don't have to.

File: vendor/github.com/hashicorp/go-cleanhttp/cleanhttp.go (hunk 1 above)
Comment (tgross) [reply vendor/github.com/hashicorp/go-cleanhttp/cleanhttp.go:35]: Good call, will do.

**Comment by tgross** [comment]
> The line between improvement vs bug seems very fuzzy to me. Is it worth backporting to 1.0.X? Though, dependency changes curry a bit of risk so it makes sense not to backport if we don't have to.

I don't have a strong opinion on that but seeing as how this was Support-facing, we should probably plan on backporting. I've modified the changelog based on that.

File: vendor/github.com/hashicorp/go-getter/client.go [hunk 2]
```diff
@@ -289,7 +298,7 @@ func (c *Client) Get() error {
 		// if we're specifying a subdir.
 		err := g.Get(dst, u)
 		if err != nil {
-			err = fmt.Errorf("error downloading '%s': %s", src, err)
+			err = fmt.Errorf("error downloading '%s': %s", u.Redacted(), err)
```
Comment (picatz) [inline vendor/github.com/hashicorp/go-getter/client.go:301]: FWIW, this line is particularly interesting (the optional insecure config aside). 🤔 

My thinking being that this method would help certain, but not all cases where credentials are included as part of the URL. Not totally sure how I feel about it, but not a reason to block the PR.

**Review by picatz — APPROVED** [review]

File: vendor/github.com/hashicorp/go-getter/client.go (hunk 2 above)
Comment (tgross) [reply vendor/github.com/hashicorp/go-getter/client.go:301]: It might be nice to introduce in other use cases in go-getter itself, for sure.

**Pull request merged**